django_cloud_deploy/cli/prompt.py

```diff
@@ -1136,6 +1138,81 @@ def _validate(self, s: str):
                               '"test@example.com"').format(s))
 
 
+class DjangoSettingsPathPrompt(StringTemplatePrompt):
+    """Allow the user to enter the settings file path of a Django project."""
+
+    PARAMETER = 'django_settings_path'
+    MESSAGE = ('{} Enter the settings file path you want to use for deployment')
+
+    def prompt(self, console: io.IO, step: str,
+               args: Dict[str, Any]) -> Dict[str, Any]:
+        """Extracts user arguments through the command-line.
+
+        Args:
+            console: Object to use for user I/O.
+            step: Message to present to user regarding what step they are on.
+            args: Dictionary holding prompts answered by user and set up
+                command-line arguments.
+
+        Returns:
+            A Copy of args + the new parameter collected.
+        """
+        new_args = dict(args)
+
+        django_directory_path = args.get('django_directory_path', None)
+        validate = functools.partial(self._validate, django_directory_path)
+        if self._is_valid_passed_arg(console, step,
+                                     args.get(self.PARAMETER, None), validate):
+            return new_args
+
+        base_message = self.MESSAGE.format(step)
+        default_settings_path = utils.guess_settings_path(django_directory_path)
+        if default_settings_path:
+            default_message = '[{}]: '.format(default_settings_path)
+            msg = '\n'.join([base_message, default_message])
+        else:
+            msg = base_message
+        answer = _ask_prompt(
+            msg, console, validate, default=default_settings_path)
+        new_args[self.PARAMETER] = answer
+        return new_args
+
+    def _validate(self, project_dir: str, django_settings_path: str):
+        """Validates that a string is a valid Django settings file path.
+
+        Args:
+            project_dir: Absolute path of the existing Django project.
+            django_settings_path: The string to validate.
+
+        Raises:
+            ValueError: if the input string is not a valid settings file path.
+        """
+        if not os.path.exists(django_settings_path):
+            raise ValueError(
+                'Path ["{}"] does not exist.'.format(django_settings_path))
+
+        if not django_settings_path.endswith('.py'):
+            raise ValueError(
+                '["{}"] is not a .py file.'.format(django_settings_path))
+
+        module_relative_path = os.path.relpath(django_settings_path,
+                                               project_dir)
+
+        # Remove ".py" at the end.
+        module_name = module_relative_path[:-3].replace('/', '.')
```

Comment:
os.path.splitext will remove the .py for you - and you can use it for the extension checking above too

Reply:
Done.